### ISSUE TITLE
[Accessibility] Checkboxes are not accessible when disabled

### ISSUE THREAD

**Issue by tallulahkay** [civiform/civiform#11922]
**Describe the bug**
When an applicant reaches the maximum number of location selections, the checkboxes are disabled and removed from the tab order. This hampers the ability for screen reader users to continue looking into other locations.

**WCAG Success Criterion**
[3.2.4 Consistent Identification (Level AA)](https://www.w3.org/TR/WCAG21/#consistent-identification)
[4.1.2 Name, Role, Value (Level A)](https://www.w3.org/TR/WCAG21/#name-role-value)

**To Reproduce**
1. Navigate to a location selection question with a maximum limit
2. Select the maximum number of allowed locations
3. Use Tab key to navigate through remaining location cards
4. Observe that the checkboxes are no longer focusable

**Expected behavior**
When checkboxes are disabled at maximum selections, the checkboxes should remain in the tab order, but there should be clear audible indication (via screen reader text) that the checkbox is currently disabled and cannot be selected
